### PR TITLE
Cyberiad Bridge remap

### PR DESCRIPTION
**Pull request by S34NW**
This really needs a TM to ensure things are actually better this way.

## What Does This PR Do
Partially ports the cyberiad bridge remap from https://github.com/ss220club/Paradise-SS220/pull/199

Moves AI upload out of bridge, placing it in science and incorporates some measures to toughen defenses.
Moves gravity generator out of bridge, placing it near engineering where it is exposed to asteroids and maintenance.
Adds NSS Cyberiad statue by @dj-34
Messaging server room is now rad-proof so bridge residents can shelter safely.
(The fish in the fish pools are immune to random sentience.)

## Why It's Good For The Game
Cyberiad bridge has long had a few issues that I've previously tried to tackle (See: https://github.com/ParadiseSS13/Paradise/pull/16406)
The main point that carries over from that concept is the AI upload location:

- Currently, AI upload is completely and utterly inaccessible to antags. A large part of this is its proximity to the bridge, anyone on the bridge is currently instantly alerted to an intruder. This encourages subverting AI via a hidden upload, and leads to a ping-pong law change battle.

Other benefits include a nicer more open feel, with more entry routes into the bridge that aren't in attack range of hobos.

Gravity generator also benefits from being in a more remote location for antag play and meteor exposure.

## Images of changes
Updated as of 2023/11/06

____

___

https://github.com/ParadiseSS13/Paradise/assets/12197162/e733f77f-02b2-4c8c-aa9e-0717f19bc05c

## Testing
Compiled and observed the map, needs playtesting with actual players.

## Changelog
:cl: dj-34, S34N
tweak: Remapped Cyberiad bridge area, moved AI upload next to vault, moved explorer room to replace abandoned shop front.
/:cl:

### CODE REVIEW

**Comment by Komrad822** [comment]
I like it but i feel the grav gen there will cause more harm than good since command use ai upload to escape rad storms at the moment, where would they run to with this change? The irradiated grav gen area? 

Or the grav gen gets disabled and has to be restarted, oh lovely the CE walks out greener than a christmas tree and nian/kiban captain keels over from the slightest rad exposure (very funny ngl).

Lastly, antag man steals rd id (or antag RD) but no longer needs to make their own discrete ai upload, just has to walk into a room thats out of sight and mind from command and sec. 

Like the ideas but i see flaws that need to be addressed or taken into consideration in case they havent already.

**Comment by Fordoxia** [comment]
AI upload is currently accessible by cutting through the gravity generator's rear wall. The gravity generator only emits radiation when it's turning on or off, so it's very risk free, no one will be looking at you. It's my favourite way in to upload!

**Comment by S34NW** [comment]
Thanks for all the comments so far, they are very useful!

To answer some points:

- HoP+Meeting room will be swapped with Captain+Teleporter
- Grav gen could be made rad-proof to give the shelter option to command
- I want to move AI upload elsewhere (near sci somewhere like cere?) and move things about again to restore the abandoned store since that is used a lot for RP shopkeeper stuff

**Comment by McRamon** [comment]
Can you remove plants from AI upload, they completely dotn fit the place

**Comment by S34NW** [comment]
Major updates to layout and AI upload implemented

**Comment by Sadhorizon** [comment]
Just to make sure, explosions from the EXPERIMENTOR will not damage the upload, correct?

**Comment by warriorstar-orion** [comment]
This looks really sick in game. Few small things:

This air alarm is probably meant to be an east bump

2023_10_27__06_39_28__paradise dme  cyberiad dmm  - StrongDMM

This tile is missing the grass patch floor

2023_10_27__06_39_38__paradise dme  cyberiad dmm  - StrongDMM

Move the light off the glass? Not really a window so up to you

2023_10_27__06_42_16__paradise dme  cyberiad dmm  - StrongDMM

**Comment by S34NW** [comment]
> Just to make sure, explosions from the EXPERIMENTOR will not damage the upload, correct?
> 
> They are rather close and EXPERIMENTOR has a tendency to blow up sometimes, primarily with the strange object bombs.
> 
> I highly doubt anything would happen but better to be sure!

Will be good to find this out in a TM

**Comment by S34NW** [comment]
> This looks really sick in game. Few small things:
> 
> This air alarm is probably meant to be an east bump
> 
> 2023_10_27__06_39_28__paradise dme cyberiad dmm - StrongDMM
> 
> This tile is missing the grass patch floor
> 
> 2023_10_27__06_39_38__paradise dme cyberiad dmm - StrongDMM
> 
> Move the light off the glass? Not really a window so up to you
> 
> 2023_10_27__06_42_16__paradise dme cyberiad dmm - StrongDMM

Many thanks, addressed

**Comment by Denthamos** [comment]
For the gravity generator it is kinda inefficient and not fitting ICly to have the radsuits inside the room which gives off the radiation instead of having a prep room like it does currently.  Is there anyway to keep a suiting up room for it with the suits in it?  That room could also be made into the radiation proof room from rad storms as a scrubbing room instead of making the entire grav gen rad proof.

Otherwise I do like how this looks.  Very fancy.

**Review by Henri215 — CHANGES_REQUESTED** [review]
1. Captain quarters is missing the captain rubber ducky
image

2. One can access that extinguisher cabinet?
image

3. This locker shouldnt be using the EAST version?
image

4. This pipe placed by accident here

image

5. No APC inside this area
image

6. Two cameras right next to each other, not necessary
image

7. Could give an organised placement to the AI modules on the upload chamber table

8. Same here, you can access that air alarm on the meeting room?
image

9. Request console placed by accident, only one is necessary and its blocking the keycard authentication device
image

**Comment by S34NW** [comment]
> 1. Captain quarters is missing the captain rubber ducky
>        image
> 
>     2. One can access that extinguisher cabinet?
>        image
> 
>     3. This locker shouldnt be using the EAST version?
>        image
> 
>     4. This pipe placed by accident here
> 
> 
> image
> 
>     5. No APC inside this area
>        image
> 
>     6. Two cameras right next to each other, not necessary
>        image
> 
>     7. Could give an organised placement to the AI modules on the upload chamber table
> 
>     8. Same here, you can access that air alarm on the meeting room?
>        image
> 
>     9. Request console placed by accident, only one is necessary and its blocking the keycard authentication device
>        image

Fixed all points bar 3. Point 3 is an issue with the east/west sprites for the cabinet being flipped.

**Comment by Bjamcham** [comment]
<details>
<summary>A few bits I'm curious about</summary>
<br>
1. Could the cap air alarm and fire alarm possibly shift over to the north and west wall just to clear it up?

2. The maints room seems like a really vulnerable bridge access point with both the bridge and bedroom walls as normal walls. Breaking into the cap's office is now only two subtle welds away.

image

3. With the new fire extinguisher sprites, would it be better to shift the cabinet one wall left in the meeting room?
image

4. Would these lights be better on the walls just north and south?
image

5. Was the meeting room intended not to have any electrified grilles? 
6. `Gravity Generator prep room`, should prep and room be capitalized?
7.  This may just be me but this section looks really crowded with wall mounts. Maybe move the grav gen fire alarm to an east/west wall, the bridge apc and air alarms could fit in on the actual wrap around hallway, exposing a potential bit of sabotage against the final bridge itself.
image

8. Would the cam be better off on a wall above or below?
image

9. This section of walls is very spartan although I'm not sure what could be added.
image

</details>

**Comment by Banderblitz** [comment]
I'll add my own two cents on this since I frequently command main and or deal with AI shenanigans.

Bridge area is nice except for two key points.

1. Engineering should still have access to the gravity generator to fix it without requiring a CE as most often there *isnt* one. Flip it around and give engineers access to the bottom two lower bridge doors and I think that is fixed.
2. Captain's bedroom has a single normal wall to break into right next his bed and the locker, I get that things being too secure is problematic, but a *single* normal wall to get the captain's locker is too easy... every single head of staff's locker is harder to get to due to being public or at LEAST a reinforced one and that should be fixed.

Now AI upload: I get the whole pingponging AI laws because of a convenient to access upload for captain to use, but making it FAR too inconvenient to use and out of the way isnt how to fix that. People that want to subvert an AI or command that want an AI upload  will still do so by getting into tech storage, it being so far out of the way in science just means less effort to try and subvert it, and significantly harder to actually fix it because it's just out of the way. The AI upload should at LEAST be as close as the RD office is to the main primary hallway now, if not closer IMO for 'security' sake since it's the AI upload it shouldn't be a remote room. Even if it requires shifting the entirety of science on the same side down. I believe it's easily doable to fit it closer while not making it so far out of the way that subverting it is as easy as breaking down one wall where noone ever is, and changing laws.

**Comment by ItsMarmite** [comment]
scientists can turn off the AI upload turret control
RD remote can't interact with the doors

**Comment by S34NW** [comment]
> A few bits I'm curious about
> 
> 1. Could the cap air alarm and fire alarm possibly shift over to the north and west wall just to clear it up?
> 
>     2. The maints room seems like a really vulnerable bridge access point with both the bridge and bedroom walls as normal walls. Breaking into the cap's office is now only two subtle welds away.
> 
> 
> image
> 
>     3. With the new fire extinguisher sprites, would it be better to shift the cabinet one wall left in the meeting room?
>        image
> 
>     4. Would these lights be better on the walls just north and south?
>        image
> 
>     5. Was the meeting room intended not to have any electrified grilles?
> 
>     6. `Gravity Generator prep room`, should prep and room be capitalized?
> 
>     7. This may just be me but this section looks really crowded with wall mounts. Maybe move the grav gen fire alarm to an east/west wall, the bridge apc and air alarms could fit in on the actual wrap around hallway, exposing a potential bit of sabotage against the final bridge itself.
>        image
> 
>     8. Would the cam be better off on a wall above or below?
>        image
> 
>     9. This section of walls is very spartan although I'm not sure what could be added.
>        image

1. Fixed
2. Changed access to atmos only, so hackers have to go through the door or weld rwall, can revisit this if it's still a problem
3. Fixed
4. Fixed
5. Yes as it has shutters
6. Good shout, fixed
7. Moved as suggested, fixed
8. Fixed
9. Fixed somewhat

File: code/game/area/ss13_areas/engineering_areas.dm [hunk 1]
```diff
@@ -50,6 +50,9 @@
 	name = "\improper Gravity Generator"
 	icon_state = "gravgen"
 
+/area/station/engineering/gravitygenerator/entry
+	name = "\improper Gravity Generator prep room"
```
Comment (S34NW) [inline code/game/area/ss13_areas/engineering_areas.dm:54]:
```suggestion
	name = "\improper Gravity Generator Prep Room"
```

**Comment by S34NW** [comment]
> I'll add my own two cents on this since I frequently command main and or deal with AI shenanigans.
> 
> Bridge area is nice except for two key points.
> 
>     1. Engineering should still have access to the gravity generator to fix it without requiring a CE as most often there _isnt_ one. Flip it around and give engineers access to the bottom two lower bridge doors and I think that is fixed.
> 
>     2. Captain's bedroom has a single normal wall to break into right next his bed and the locker, I get that things being too secure is problematic, but a _single_ normal wall to get the captain's locker is too easy... every single head of staff's locker is harder to get to due to being public or at LEAST a reinforced one and that should be fixed.
> 
> 
> Now AI upload: I get the whole pingponging AI laws because of a convenient to access upload for captain to use, but making it FAR too inconvenient to use and out of the way isnt how to fix that. People that want to subvert an AI or command that want an AI upload will still do so by getting into tech storage, it being so far out of the way in science just means less effort to try and subvert it, and significantly harder to actually fix it because it's just out of the way. The AI upload should at LEAST be as close as the RD office is to the main primary hallway now, if not closer IMO for 'security' sake since it's the AI upload it shouldn't be a remote room. Even if it requires shifting the entirety of science on the same side down. I believe it's easily doable to fit it closer while not making it so far out of the way that subverting it is as easy as breaking down one wall where noone ever is, and changing laws.

1. Is a valid point, I'm going to have to think more about how we can make this work better, thank you.
2. I've mitigated this with removing maint access and applying atmos access to the door, if this continues to be an issue I will revisit the wall setup
3. AI upload is going to be fine here, on Cere it's even further from RD and much more remote, but is still accessible and has worked perfectly fine there. This implementation trades some of the security of the cere upload (double rwall) for proximity to people. There is almost always someone in science at any given point so it gets security on that front.

**Comment by S34NW** [comment]
> scientists can turn off the AI upload turret control
> RD remote can't interact with the doors

1. This was an issue from copy pasting the turret controller, likely exists as a bug on current cyberiad just not noticed by anyone. Fixed.
2. This is the same as cyberiad, only command remote has access to the door. Tweaks to the remote needed for that.

**Comment by Freakycrafter** [comment]
grafik
The bridge's power monitor isnt connected yet.

**Comment by S34NW** [comment]
> grafik The bridge's power monitor isnt connected yet.

Fixed

**Comment by S34NW** [comment]
Major changes to the remap implemented, see images.

**Comment by go-liath-online** [comment]
Loving this remap so far. A few nitpicks:

- AI upload has two holopads, which is probably unnecessary for such a small room. Even atmos only has one!
- AI upload is missing AI status displays. Might flesh the room out a bit better.
- AI upload intercom is on General, whereas in the old upload, it was on Private.
- Gravity generator holopad is in a strange spot.
- The new placement of the camera aft of the bridge means AI has a blind spot between the Cyberiad sculpture and the aft hallway.
- The intercom in the meeting room isn't in the middle of the table anymore.
- The red phone in the heads of staff meeting room is in Russian.

**Comment by S34NW** [comment]
> Loving this remap so far. A few nitpicks:
> 
>     * AI upload has two holopads, which is probably unnecessary for such a small room. Even atmos only has one!
> 
>     * AI upload is missing AI status displays. Might flesh the room out a bit better.
> 
>     * AI upload intercom is on General, whereas in the old upload, it was on Private.
> 
>     * Gravity generator holopad is in a strange spot.
> 
>     * The new placement of the camera aft of the bridge means AI has a blind spot between the Cyberiad sculpture and the aft hallway.
> 
>     * The intercom in the meeting room isn't in the middle of the table anymore.
> 
>     * The red phone in the heads of staff meeting room is in Russian.

Thanks, a fix is in progress for all of these points.

**Comment by matttheficus** [comment]
Personal nitpicks:

- This double-wall looks off to me. Not a fan of double-walls to begin with.
image

- This entire hallways is now a nightmare of MetaStation proportions - it is for all intents a 1-tile wide hallway and make a significant chokepoint in SciMaints compared to the usual version with the "spare room". Getting cornered here is a death sentence.
image

- Some of the Bridge doors are missing firelocks.

**Comment by matttheficus** [comment]
image

The turrets in the Upload are now half useless. They cannot hit anyone walking into the doors, and if set to lethal, will just break things.

**Comment by matttheficus** [comment]
The Bridge APC now runs out of power VERY VERY VERY quickly before the Engine/SMESs are set up - even faster than Cargo's APC.

**Comment by S34NW** [comment]
> Personal nitpicks:
> 
>     * This double-wall looks off to me. Not a fan of double-walls to begin with.
>       image
> 
>     * This entire hallways is now a nightmare of MetaStation proportions - it is for all intents a 1-tile wide hallway and make a significant chokepoint in SciMaints compared to the usual version with the "spare room". Getting cornered here is a death sentence.
>       image
> 
>     * Some of the Bridge doors are missing firelocks.

Addressed



> The Bridge APC now runs out of power VERY VERY VERY quickly before the Engine/SMESs are set up - even faster than Cargo's APC.

Addressed



> image
> 
> The turrets in the Upload are now half useless. They cannot hit anyone walking into the doors, and if set to lethal, will just break things.

Bit of a tricky one, but attempted to address.

**Comment by go-liath-online** [comment]
Heads up, you have `/obj/machinery/status_display/supply_display` in the AI upload instead of `/obj/machinery/ai_status_display` so the screen in there just shows the cargo shuttle timer.

**Review by Git-GoR — CHANGES_REQUESTED** [review]
<details><summary>Some issues in Bridge area:</summary>

1.  Teleporter room missing the yellow caution stripes.
imagen

1. In the eastern SE Central Hallway you've added a fire alarm, when there was already one a few tiles north, right before the corner.
imagen

1. HoP's Office Door button needs to be linked to the airlock.
imagen

1. The Privacy Shutters on the table of HoP's office have a `dir` 8, making them close the other way around of the rest in the same line.
imagen

1. Even with night lighting off, the south entrance of the Bridge Lounge is kinda dark. It could use a lightbulb. This would also help greatly with the visibility in the lower Lounge during night lights as a whole, as currently is sort of dark.
imagen

1. Couple of camera darkspots in the Lounge south entrance. Not the biggest deal, but could be a small nuisance as AI. (I wonder if non room wide darkspots would be a mechanic worth adding to other areas)
imagen

1. Lounge has a very low intercom coverage.
imagen

1. Meeting room has no fire alarm.
imagen

1. Lounge has no fire extinguisher cabinets (Greygang buff :alien:)
</details>

<details><summary>Single thing in Gravgen:</summary>

1. The Gravity Generator SMES is now connected to the general powernet throught the input, intended? While this will not impact it with a powersink, it might add a bit of power requirement to the station as a whole as it's another SMES that needs to be charged.
imagen

</details>

<details><summary>Couple things in RnD Area:</summary>

1. Experimentor room lacks the button to close the protective shutters.
imagen

1. The Camera console and the camera itself are added to the network... `Telepad`? The previous one was in the `Experimientator` net, judging by the prefabs. It works, but makes little sense.
imagen

</details>

**Comment by S34NW** [comment]
> Some issues in Bridge area:
> 
>     1. Teleporter room missing the yellow caution stripes.
>        imagen
> 
>     2. In the eastern SE Central Hallway you've added a fire alarm, when there was already one a few tiles north, right before the corner.
>        imagen
> 
>     3. HoP's Office Door button needs to be linked to the airlock.
>        imagen
> 
>     4. The Privacy Shutters on the table of HoP's office have a `dir` 8, making them close the other way around of the rest in the same line.
>        imagen
> 
>     5. Even with night lighting off, the south entrance of the Bridge Lounge is kinda dark. It could use a lightbulb. This would also help greatly with the visibility in the lower Lounge during night lights as a whole, as currently is sort of dark.
>        imagen
> 
>     6. Couple of camera darkspots in the Lounge south entrance. Not the biggest deal, but could be a small nuisance as AI. (I wonder if non room wide darkspots would be a mechanic worth adding to other areas)
>        imagen
> 
>     7. Lounge has a very low intercom coverage.
>        imagen
> 
>     8. Meeting room has no fire alarm.
>        imagen
> 
>     9. Lounge has no fire extinguisher cabinets (Greygang buff 👽)
> 
> 
> Single thing in Gravgen:
> 
>     1. The Gravity Generator SMES is now connected to the general powernet throught the input, intended? While this will not impact it with a powersink, it might add a bit of power requirement to the station as a whole as it's another SMES that needs to be charged.
>        imagen
> 
> 
> Couple things in RnD Area:
> 
>     1. Experimentor room lacks the button to close the protective shutters.
>        imagen
> 
>     2. The Camera console and the camera itself are added to the network... `Telepad`? The previous one was in the `Experimientator` net, judging by the prefabs. It works, but makes little sense.
>        imagen

All points fixed. Although as a note, the cam console and experimentor cam net were unchanged by me. Not sure what happened there 😕

**Review by Git-GoR — CHANGES_REQUESTED** [review]
All good, just one small thing. Due to the junction to HoP's office, the Meeting Room disposals will throw all the trash back out inmediately.
imagen

Could be solved by doing something like this, or whatever you come up with (disposals pain):
imagen

Once that is fixed, it should be good to go.

**Comment by S34NW** [comment]
> All good, just one small thing. Due to the junction to HoP's office, the Meeting Room disposals will throw all the trash back out inmediately.
> imagen
> 
> Could be solved by doing something like this, or whatever you come up with (disposals pain):
> imagen
> 
> Once that is fixed, it should be good to go.
> 

Have you confirmed this? It was my understanding that trash hitting specifically sort junctions from the reverse direction will pass as expected. This is the behaviour in a number of places on-station as per my understanding.

**Comment by Git-GoR** [comment]
I was not sure myself as well, so I tested it by chucking a random item from the room into the chute, which bounced out one second later. I tried disposaling myself in, same result.

I did not test a tagged item, maybe in that case works different? In any case, please double check just to fully confirm it.

**Comment by S34NW** [comment]
> I was not sure myself as well, so I tested it by chucking a random item from the room into the chute, which bounced out one second later. I tried disposaling myself in, same result.
> 
> I did not test a tagged item, maybe in that case works different? In any case, please double check just to fully confirm it.

Fixed

**Comment by Contrabang** [comment]
image
This APC is disconnected

**Comment by Henri215** [comment]
Its looking good to me, just need to fix that APC issue

**Comment by S34NW** [comment]
> image This APC is disconnected

fixed

**Review by Henri215 — APPROVED** [review]

**Review by hal9000PR — APPROVED** [review]

**Review by Git-GoR — APPROVED** [review]

**Review by lewcc — APPROVED** [review]
LGTM (let's get this mapped)

**Pull request merged**